engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/atom/AtomConstructor.java

```diff
@@ -359,4 +359,14 @@ boolean hasType() {
   Type getType(@CachedLibrary("this") TypesLibrary thisLib, @Cached("1") int ignore) {
     return EnsoContext.get(thisLib).getBuiltins().function();
   }
+
+  @ExportMessage
+  boolean hasMetaObject() {
+    return true;
+  }
+
+  @ExportMessage
+  Type getMetaObject(@CachedLibrary("this") InteropLibrary thisLib, @Cached("1") int ignore) {
```

Comment:
What kind of magical trick does the `@Cached("1") int ignore` parameter?

Reply:
FYI: https://github.com/enso-org/enso/pull/7493/commits/9659ca01dd7747d8c629d3c29e52a7650223a231#r1286876004